### PR TITLE
[aclorch]: Add MIRRORv6 support for ACL table

### PR DESCRIPTION
**Pull request by stcheng**
Enable both IPv4 and IPv6 source and destination IP match and
mirror functionality in SONiC.

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

### CODE REVIEW

File: orchagent/aclorch.cpp [hunk 1]
```diff
@@ -948,6 +956,20 @@ bool AclRuleMirror::validateAddMatch(string attr_name, string attr_value)
         return false;
     }
 
+    if (m_tableType == ACL_TABLE_MIRROR &&
+            (attr_name == MATCH_SRC_IPV6 || attr_name == MATCH_DST_IPV6))
+    {
+        SWSS_LOG_ERROR("shuotian: does not match!!!");
```
Comment (jleveque) [inline orchagent/aclorch.cpp]: Did you intend to leave these log messages which contain your name?

File: tests/test_mirror_ipv6.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        # assert acl rule is created
+        tbl = swsscommon.Table(self.adb, "ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY")
+        rule_entries = [k for k in tbl.getKeys() if k not in dvs.asicdb.default_acl_entries]
+        assert len(rule_entries) == 1
+
+        (status, fvs) = tbl.get(rule_entries[0])
+        assert status == True
+        for fv in fvs:
+            if fv[0] == "SAI_ACL_ENTRY_ATTR_FIELD_SRC_IPV6":
+                assert fv[1] == "2777::&mask:ffff:ffff:ffff:ffff::"
```
Comment (qiluo-msft) [inline tests/test_mirror_ipv6.py]: '&mask' is weird. Could you explain?

Comment (stcheng) [reply tests/test_mirror_ipv6.py]: this is the serialization done by sairedis library; swss does not have the control here.

**Comment by stcheng** [comment]
Updated with new design

**Comment by stcheng** [comment]
TODO:
1. will update the unit test both both Broadcom and Mellanox scenarios
2. will remove deprecated code in swss for virtual switch platform

File: orchagent/aclorch.cpp [hunk 3]
```diff
@@ -948,6 +954,43 @@ bool AclRuleMirror::validateAddMatch(string attr_name, string attr_value)
...
+     * | MATCH_DST_IPV6   |      √       |              |       √        |
+     * |-----------------------------------------------------------------|
+     * | MARTCH_ETHERTYPE |      √       |      √       |                |
+     * |-----------------------------------------------------------------|
+     */
+
+    if (m_tableType == ACL_TABLE_MIRROR && !m_pAclOrch->isCombinedMirrorV6Table() &&
+            (attr_name == MATCH_SRC_IPV6 || attr_name == MATCH_DST_IPV6))
+    {
+        return false;
```
Comment (lguohan) [inline orchagent/aclorch.cpp]: why not have syslog error message here?

File: orchagent/aclorch.cpp [hunk 4]
```diff
@@ -948,6 +954,43 @@ bool AclRuleMirror::validateAddMatch(string attr_name, string attr_value)
...
+    {
+        SWSS_LOG_ERROR("%s match is not supported for the table of type MIRRORV6",
+                attr_name.c_str());
+        return false;
+    }
+
+    if (m_tableType == ACL_TABLE_MIRRORV6 &&
+            (attr_name == MATCH_SRC_IP || attr_name == MATCH_DST_IP || attr_name == MATCH_ETHER_TYPE))
+    {
+        return false;
```
Comment (lguohan) [inline orchagent/aclorch.cpp]: why not have syslog error message here?

File: orchagent/aclorch.cpp [hunk 5]
```diff
@@ -948,6 +954,43 @@ bool AclRuleMirror::validateAddMatch(string attr_name, string attr_value)
...
+    }
```
Comment (lguohan) [inline orchagent/aclorch.cpp]: is this duplicated code?

File: orchagent/aclorch.cpp [hunk 6]
```diff
@@ -1780,6 +1860,64 @@ void AclOrch::init(vector<TableConnector>& connectors, PortsOrch *portOrch, Mirr
...
+                fvVector.emplace_back(TABLE_TYPE_MIRROR, value);
+                break;
+            case ACL_TABLE_MIRRORV6:
+                fvVector.emplace_back(TABLE_TYPE_MIRRORV6, value);
+                break;
+            default:
+                break;
+        }
+    }
+    m_switchTable.set("switch", fvVector);
```
Comment (lguohan) [inline orchagent/aclorch.cpp]: this code really need to go to syncd code. We will address this in the next release.

File: orchagent/aclorch.cpp [hunk 7]
```diff
@@ -2152,7 +2348,14 @@ void AclOrch::doAclRuleTask(Consumer &consumer)
                 continue;
             }
 
-            newRule = AclRule::makeShared(m_AclTables[table_oid].type, this, m_mirrorOrch, m_dTelOrch, rule_id, table_id, t);
+            auto type = m_AclTables[table_oid].type;
+            if (type == ACL_TABLE_MIRROR || type == ACL_TABLE_MIRRORV6)
+            {
+                    type = table_id == m_mirrorTableId ? ACL_TABLE_MIRROR : ACL_TABLE_MIRRORV6;
```
Comment (lguohan) [inline orchagent/aclorch.cpp]: identation

File: orchagent/aclorch.cpp (hunk 6 above)
Comment (stcheng) [reply orchagent/aclorch.cpp]: sure i could add some todo to here

Comment (stcheng) [reply orchagent/aclorch.cpp]: One issue about moving this to syncd right now is that we currently cannot guarantee the order of syncd and orchagent starting time. e.g. when orchagent starts and at that time syncd hasn't stored the capability information into the database, we will miss this part of the information.

**Pull request merged**